compiler/environment.ml

```diff
@@ -0,0 +1,116 @@
+open Ast;;
+
+module FunctionMap = Map.Make(String);;
+module VariableMap = Map.Make(String);;
+module ArrayTypeMap = Map.Make(String);;
+
+exception VarAlreadyDeclared;;
+exception VarNotDeclared;;
+exception FunctionAlreadyDeclared;;
+exception FunctionNotDeclared;;
+exception IncorrectFunctionParameterTypes;;
+exception MixedTypeArray;;
+exception ArrayLiteralTypeMismatch;;
+
+type func_info  = {
+  id : string; 
+  return : data_type; 
+  args : data_type list;
+  arg_names: string list;
+}
+
+type symbol_table = {
+  func_map: func_info FunctionMap.t;
+  var_map: data_type VariableMap.t;
+  array_type_map: data_type ArrayTypeMap.t;
+}
+
+let create = 
+  {
+    func_map = FunctionMap.empty;
+    var_map = VariableMap.empty;
+    array_type_map = ArrayTypeMap.empty;
+  }
+
+let update f_map v_map a_type_map =
```

Comment:
Totally hear where you are coming from, but I think it's ok in this instance, since the utility is to update all the maps and that's it.  So it's not really doing a whole bunch, so I think it is not a bad idea to abstract it to one function